src/monattack.cpp

```diff
@@ -1919,6 +1997,9 @@ void mattack::fear_paralyze(monster *z, int index)
 
 void mattack::photograph(monster *z, int index)
 {
+    if( z->friendly ) {
+        return; // TODO: handle friendly monsters
```

Comment:
Friendly eyebots should probably not do anything.  It's possible you might have one join you to monitor a quest or something similar, so having 'em disappear wouldn't work.